File: app/src/java/java/com/example/ama/viewer/di/modules/UtilsModule.java
package com.example.ama.viewer.di.modules;

import android.content.Context;

import com.example.ama.viewer.utils.ResUtils;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class UtilsModule {

    @Provides
    @Singleton
    ResUtils provideResUtils(Context appContext) {
        return new ResUtils(appContext);
    }
}

File: app/src/kotlin/java/com/example/ama/viewer/di/modules/UtilsModule.kt
package com.example.ama.viewer.di.modules

import android.content.Context
import com.example.ama.viewer.utils.ResUtils
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class UtilsModule {

    @Provides
    @Singleton
    fun provideResUtils(app: Context): ResUtils = ResUtils(app)
}